### PR TITLE
Hotfix/bigfoot fixing an allocation assumption

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [n/a ] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [n/a] `ltnewsX.tex` updated

@josephwright once merged can you upload firstaid?

I will afterwards do the integration into develop

### CODE REVIEW

**Review by josephwright — APPROVED** [review]

File: required/firstaid/latex2e-first-aid-for-external-files.dtx [hunk 1]
```diff
@@ -397,6 +397,45 @@
...
+%
+%    This firstaid therefore jumps over the problematical point by
+%    pushing the count allocation to a safe value if necessary.
+%    \begin{macrocode}
+\AddToHook{file/before/bigfoot.sty}{%
+   \ifnum\count10>188 %
+   \else
+     \global\count10=189 %
+   \fi
+%    \end{macrocode}
```
Comment (davidcarlisle) [inline required/firstaid/latex2e-first-aid-for-external-files.dtx:420]: am I missing something here, but why 188 and why just check count 10?

`\newinsert` switches to stealling boxes from the float free list if

```
\ifnum\count10<\insc@unt
\ifnum\count11<\insc@unt
\ifnum\count12<\insc@unt
\ifnum\count14<\insc@unt
```

so don't you need to make count10 bigger than `insc@ount` (which is 198 initially with current kernel allocations)

Comment (FrankMittelbach) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:420]: maybe you do  or maybe I do :-).  Point is, perpage, manyfoot and bigfoot (which loads the other two) do allocations and there is no reason to push them out. 198 is bigger than 188 last time I looked, so making a tests for 198 and pushing to that would make a much bigger gap than just going to 189 which is enough to get the first important insert taken from the float list. So basically what the test does is to ensure that the count register allocation is at least high enough that inserts will always be stolen from the floats. As count registers are close to that cutoff value that is all that is needed (assuming perpage/manyfoot/bigfoot are not suddenly dropping register allocations (which is rather unlikely). Anything wrong with that reasoning?

Comment (davidcarlisle) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:420]: er, I suppose so OK let this go through it's only first-aid after all, the hardcoded 188 looked slightly strange, but it  is essentially \insc@ount - number of \newcount bigfoot does before it starts doing \newinsert, so would be wrong if another package does inserts and changes inscount, that is pretty unlikely though, to load another insert packeg. I guess \numexpr\insc@ount-10\relax would be safer?

Comment (FrankMittelbach) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:420]: I agree the value is a but odd, but if there are other packages that allocate stuff it doesn't get wrong as far as I can see, as it is only a minimal value and so would then not apply and all is still good. But I guess going after the package is probably clearer
```
\AddToHook{file/after/bigfoot.sty}{%
   \ifnum\count10<\insc@unt %
     \global\count10=\insc@unt %
   \fi}
```

Comment (josephwright) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:419]: Looks nice to me [other than we could drop those `%` as they are redundant ;)]

Comment (FrankMittelbach) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:419]: yes and no. They are redundent, but they point out that the space is deliberate and not by mistake.

Comment (FrankMittelbach) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:419]: so are we good with that version @davidcarlisle ?

**Review by davidcarlisle — APPROVED** [review]

File: required/firstaid/latex2e-first-aid-for-external-files.dtx (hunk 1 above)
Comment (josephwright) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:419]: @FrankMittelbach  My point was the version with `\ifnum\count10<\insc@unt %` doesn't need the `%` - no big deal, just a bit of fun really.

Comment (josephwright) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:419]: I'll wait for a merge to main then do a release of first-aid.

Comment (FrankMittelbach) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:419]: > @FrankMittelbach My point was the version with `\ifnum\count10<\insc@unt %` doesn't need the `%` - no big deal, just a bit of fun really.

right you are ... no longer

**Pull request merged**